packages/moonstone/styles/variables.less

```diff
@@ -430,9 +433,14 @@
 
 // Dialog
 // ---------------------------------------
-@moon-dialog-top-padding: 24px;
+@moon-dialog-top-padding: 42px;
 @moon-dialog-h-padding: 42px;
-@moon-dialog-bottom-padding: 42px;
+@moon-dialog-bottom-padding: 24px;
+@moon-dialog-reserve-close-right-padding: (@moon-button-h-margin * 2) + @moon-icon-button-small-size + @moon-dialog-bottom-padding;
```

Comment:
I think this sort of refactoring is immensely useful. Using other components variables as a base for calculations is great for visual consistency but often easy to overlook when changing them later. By consolidating those relationships into the variable files, I think those relationships are easier to identify. :+1: :+1: